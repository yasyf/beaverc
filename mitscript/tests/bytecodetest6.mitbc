function
{
	functions =
	[
		function
		{
			functions =
			[
				function
				{
					functions = [],
					constants = [None],
					parameter_count = 1,
					local_vars = [y],
					local_ref_vars = [],
					free_vars = [print],
					names = [],
					instructions = 
					[
						load_local	0
						load_ref	0
						call	1
						pop
						gc
						load_const	0
						return
					]
				},
				function
				{
					functions = [],
					constants = [None],
					parameter_count = 1,
					local_vars = [z],
					local_ref_vars = [],
					free_vars = [],
					names = [],
					instructions = 
					[
						load_const	0
						return
					]
				}
			],
			constants = ["STRING2", None],
			parameter_count = 1,
			local_vars = [x, f2, print],
			local_ref_vars = [print],
			free_vars = [],
			names = [],
			instructions = 
			[
				push_ref	0
				load_func	0
				alloc_closure	1
				store_local	1
				load_const	0
				load_local	1
				call	1
				pop
				gc
				load_func	1
				alloc_closure	0
				store_ref	0
				load_const	1
				return
			]
		}
	],
	constants = ["STRING2", 0],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [print, input, intcast, f],
	instructions = 
	[
		load_func	-1
		store_global	0
		load_func	-2
		store_global	1
		load_func	-3
		store_global	2
		load_func	0
		alloc_closure	0
		store_global	3
		load_const	0
		load_global	3
		call	1
		pop
		gc
		load_const	1
		return
	]
}