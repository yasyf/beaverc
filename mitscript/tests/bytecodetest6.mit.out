IllegalCastException: Can't cast this value to a pointer type.
